overlord/snapstate/snapmgr.go

```diff
@@ -224,3 +228,19 @@ func (m *SnapManager) Wait() {
 func (m *SnapManager) Stop() {
 	m.runner.Stop()
 }
+
+// SnapInfo returns snap.Info for a given snap name and snap version.
+func SnapInfo(state *state.State, snapName, snapVersion string) (*snap.Info, error) {
+	fname := filepath.Join(dirs.SnapSnapsDir, snapName, snapVersion, "meta", "snap.yaml")
+	yamlData, err := ioutil.ReadFile(fname)
+	if err != nil {
+		return nil, err
+	}
+	info, err := snap.InfoFromSnapYaml(yamlData)
+	if err != nil {
+		return nil, err
+	}
+	// XXX: requested by pedronis
```

Comment:
// Overwrite the name which doesn't belong in snap.yaml and is actually defined by
// snap declaration assertion.

^^ @pedronis is that what you meant?